Adding support for new Lutron RGB tape light

RBaragona opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Add support for new device type in Lutron-Caseta to enable support of new color and tunable white products.

Supported by integration via #134906

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: https://github.com/gurumitts/pylutron-caseta/pull/177
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] I have followed the [perfect PR recommendations][perfect-pr]
- [ ] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

RBaragona committed:
Adding support for new Lutron color tuning device, Supported via https://github.com/gurumitts/pylutron-caseta/pull/177

eclair4151 commented on homeassistant/components/lutron_caseta/light.py:
```diff
@@ -92,6 +98,8 @@ def __init__(self, light: dict[str, Any], data: LutronCasetaData) -> None:
         self.supports_warm_cool = light_type in WARM_DEVICE_TYPES
         self.supports_warm_dim = light_type == DEVICE_TYPE_SPECTRUM_TUNE
         self.supports_spectrum_tune = light_type == DEVICE_TYPE_SPECTRUM_TUNE
+        self.supports_warm_dim = light_type == DEVICE_TYPE_COLOR_TUNE
```
Won't these overwrite the lines above it? Ie if the light is spectrum tune the variables are set to true in the lines above this, then set back to false here because it is not a color_tune

RBaragona replied:
now updated!

eclair4151 commented:
Also to confirm, you'll need to wait until your pylutron-caseta PR Is merged, then raise 2 PRs here, the first just bumping the Python library version, and this is the second PR which can be merged after the first is merged.

RBaragona commented:
I agree and will link once the pylutron-caseta PR is merged.

RBaragona committed:
Update light.py

Update logic based on device types supporting warm dim and spectrum tune color.

RBaragona committed:
Merge branch 'dev' into Support-new-color-control-device

eclair4151 commented:
@RBaragona I raised the PR for bumping the pylutron version
https://github.com/home-assistant/core/pull/134906

RBaragona committed:
Merge branch 'home-assistant:dev' into Support-new-color-control-device

RBaragona committed:
Ruff updates

RBaragona committed:
Merge branch 'home-assistant:dev' into Support-new-color-control-device

RBaragona commented:
@eclair4151 , I have tested and verified control with hardware as well.

RBaragona committed:
Merge branch 'dev' into Support-new-color-control-device

joostlek commented on homeassistant/components/lutron_caseta/light.py:
```diff
@@ -90,8 +103,14 @@ def __init__(self, light: dict[str, Any], data: LutronCasetaData) -> None:
         self.supports_warm_cool = light_type in WARM_DEVICE_TYPES
-        self.supports_warm_dim = light_type == DEVICE_TYPE_SPECTRUM_TUNE
-        self.supports_spectrum_tune = light_type == DEVICE_TYPE_SPECTRUM_TUNE
+        self.supports_warm_dim = light_type in (
+            DEVICE_TYPE_SPECTRUM_TUNE,
+            DEVICE_TYPE_COLOR_TUNE,
+        )
+        self.supports_spectrum_tune = light_type in (
+            DEVICE_TYPE_SPECTRUM_TUNE,
+            DEVICE_TYPE_COLOR_TUNE,
+        )
```
```suggestion
        self.supports_warm_dim = self.supports_spectrum_tune = light_type in (
            DEVICE_TYPE_SPECTRUM_TUNE,
            DEVICE_TYPE_COLOR_TUNE,
        )
```

RBaragona replied:
This was a good catch, thanks for calling it out. It should be warm cool And warm dim support that are combined. Updated in the commit below

RBaragona committed:
Merge branch 'home-assistant:dev' into Support-new-color-control-device

RBaragona committed:
Combine support for warmcool and warmdim

eclair4151 commented on homeassistant/components/lutron_caseta/light.py:
```diff
@@ -89,9 +102,13 @@ def __init__(self, light: dict[str, Any], data: LutronCasetaData) -> None:
-        self.supports_warm_dim = light_type == DEVICE_TYPE_SPECTRUM_TUNE
-        self.supports_spectrum_tune = light_type == DEVICE_TYPE_SPECTRUM_TUNE
+        self.supports_warm_cool = self.supports_warm_dim = (
+            light_type in WARM_DEVICE_TYPES
```
This is fine with me, but when I originally wrote the PR, I was told not to let DEVICE_TYPE_WHITE_TUNE support Warm Dim this way, because Warm Dim should be controlled by a custom device feature, not using the built in 'white mode' as that was meant for something else. 

I do have plans to raise another PR to move warm dim from white mode into its own feature, but if we are ok with making DEVICE_TYPE_WHITE_TUNE (light with temp control only, not RGB) supporting warwm dim via white mode then this is fine.

RBaragona replied:
Based on that previous feedback and your planned future work, I'd like to propose I roll this back. Any concerns @joostlek ?

RBaragona committed:
Reverse changes based on comments/past feedback and planned future work

RBaragona committed:
Merge branch 'dev' into Support-new-color-control-device

RBaragona commented:
@eclair4151 and @joostlek anything else I can do to move this PR forward?

eclair4151 commented:
Everything good in my end. Happy to discuss the warm dim comment I made above with Joost if he has any thoughts. I'm fine with leaving that either way just wanted to make sure we are all in the loop.
